### PR TITLE
Refactor Team page

### PR DESCRIPTION
**Pull request by jakob223**
Note that just the 2 wip commits and the `refactor teams page` commit are the relevent ones. I'll rebase things to make this cleaner tomorrow morning.

cc @keybase/y2ksquad

### CODE REVIEW

**Comment by jakob223** [comment]
Hmm seems like this will be annoying to rebase before the sectionlist typing PR is merged.

File: shared/teams/team/container.tsx [hunk 1]
```diff
@@ -1,128 +1,45 @@
...
-
-const mapDispatchToProps = (dispatch: Container.TypedDispatch) => ({
-  onBack: () => dispatch(RouteTreeGen.createNavigateUp()),
-})
+type OwnProps = Container.RouteProps<{teamID: Types.TeamID; initialTab?: Types.TabKey}>
 
+// TODO: when the teamsRedesign flag is removed, get rid of this connector entirely and
+// move the route prop handling and the navigation options into index.tsx
+const maybeHeaderHoc = flags.teamsRedesign ? a => a : Kb.HeaderHoc
 const Connected = Container.compose(
```
Comment (buoyad) [inline shared/teams/team/container.tsx]: Seeing this error in the console:
```
mapStateToProps() in Connect(Team) must return a plain object. Instead received undefined.
```

File: shared/teams/team/index.tsx [hunk 2]
```diff
@@ -5,82 +5,175 @@ import * as Types from '../../constants/types/teams'
...
-      <TeamDetailsSubscriber teamID={props.teamID} />
-      {Styles.isMobile && flags.teamsRedesign && <MobileHeader teamID={props.teamID} offset={offset} />}
+      {Styles.isMobile && flags.teamsRedesign && <MobileHeader teamID={teamID} offset={offset} />}
       <SectionList
-        renderItem={renderItem}
         renderSectionHeader={renderSectionHeader}
         stickySectionHeadersEnabled={Styles.isMobile}
-        sections={props.sections}
-        style={styles.list}
+        sections={sections}
```
Comment (buoyad) [inline shared/teams/team/index.tsx]: The style previously pushed the list down so the header was visible, it covers the header now.

File: shared/teams/team/index.tsx [hunk 3]
```diff
@@ -5,82 +5,175 @@ import * as Types from '../../constants/types/teams'
...
+      sections.push(...invitesSections)
+      break
+    case 'settings':
+      sections.push({data: ['settings'], key: 'settings', renderItem: () => <Settings teamID={teamID} />})
+      break
+    case 'channels':
+      sections.push(...channelsSections)
+      break
+    case 'subteams':
+      sections.push(...subteamsSections)
```
[the remark this answers is not in the document]
Comment (jakob223) [reply shared/teams/team/index.tsx]: Hmm, they aren't. I wonder if I messed up in merge or if I forgot to check on this tab :/

File: shared/common-adapters/section-list.desktop.tsx [hunk 4]
```diff
@@ -236,7 +236,7 @@ class SectionList<T extends Section<any, any>> extends React.Component<Props<T>,
       <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true} style={styles.container}>
         {this.props.disableAbsoluteStickyHeader && stickyHeader}
         <Kb.ScrollView
-          style={Styles.collapseStyles([styles.scroll, this.props.contentContainerStyle])}
+          style={Styles.collapseStyles([styles.scroll, this.props.style, this.props.contentContainerStyle])}
```
Comment (buoyad) [inline shared/common-adapters/section-list.desktop.tsx]: Desktop scrollview has `contentContainerStyle`, can we use that? and change the comment in `d.ts` about it being mobile only.

File: shared/teams/team/container.tsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ type OwnProps = Container.RouteProps<{teamID: Types.TeamID; initialTab?: Types.T
 const maybeHeaderHoc = flags.teamsRedesign ? a => a : Kb.HeaderHoc
 const Connected = Container.compose(
   Container.connect(
-    () => {},
+    () => ({}),
```
Comment (buoyad) [inline shared/teams/team/container.tsx]: ah!

**Pull request merged**